### PR TITLE
doc improvements

### PR DESCRIPTION
**Pull request by adrienbernede**
Created on behalf of @rhornung67.

I’d like to see those great improvement merged.

Doc preview: https://radiuss-shared-ci.readthedocs.io/en/task-doc-improvements/

But first:
- [x] I will fix some things (doc link and trailing white spaces)
- [x] We should rehabilitate `docs/sphinx/user_guide/use_spack.rst`. (What is the interrogation about this?)
- [x] Any other changes that should be done right away? (A subsequent PR is always possible.)

### CODE REVIEW

**Comment by rhornung67** [comment]
@adrienbernede I plan to proofread the rest and make other changes. This is as far as I got.

**Comment by adrienbernede** [comment]
Great!
We can make incremental merges if you think you’ll not find the time soon. I think those changes are already beneficial.
Although I can imagine you are using this as a reminder about where you were at...

**Comment by adrienbernede** [comment]
@rhornung67 
Yes that's that plan... they come from the RADIUSS CI documentation.
I put that on the back burner, but I will resume the work at some point.

**Comment by rhornung67** [comment]
@adrienbernede I'm done making changes on this for now. If you want to review and accept or reject them, please do so. Thank you.

File: docs/sphinx/user_guide/setup_ci.rst [hunk 1]
```diff
@@ -19,25 +19,27 @@ Setup the CI using the shared template
...
+`share Spack configuration files`_ across projects. Then, we developed a 
+process to manage Spack and use it to generate CMake configuration files.
+The RADIUSS Shared CI project extends this by developing a CI implementation
+that can be shared across projects, which drive it with a ``build-and-test`` 
+script that has the same inputs across projects.
 
-By externalizing the CI configuration, we create the need for an interface.
-We try to keep this interface minimalistic, while allowing customization.
+With a centralized CI configuration that is shared by projects, we try to keep 
+its interface minimal, while allowing for project-specific customization.
```
Comment (adrienbernede) [inline docs/sphinx/user_guide/setup_ci.rst:42]:
```suggestion
With a centralized CI configuration that is shared by projects, we create an interface between local and shared configuration. we try to keep this interface minimal, while allowing for project-specific customization. Only a handful of modifications are required to get the CI to work for your projects, while files in the ``customization`` directory allow for finer tuning and extensibility.
```

File: docs/sphinx/user_guide/setup_ci.rst [hunk 2]
```diff
@@ -111,20 +113,20 @@ to add CI to.
...
-In ``.gitlab-ci.yml``, there are some variables you will need to adapt to your
-projects:
+In the ``.gitlab-ci.yml`` file, there are some variables you need to adapt to 
+your project. They are described in the following table:
 
  ========================================== ==========================================================================================================================
   Parameter                                  Description
  ========================================== ==========================================================================================================================
-  ``LLNL_SERVICE_USER``                      Service Account used in CI
+  ``LLNL_SERVICE_USER``                      Service Account for project used in CI
```
Comment (adrienbernede) [inline docs/sphinx/user_guide/setup_ci.rst:129]:
```suggestion
  ``LLNL_SERVICE_USER``                      Project specific Service Account used in CI
```

File: docs/sphinx/user_guide/setup_ci.rst [hunk 3]
```diff
@@ -190,33 +193,35 @@ details can be found in the file itself.
    If a variable is blank in the template file, then it does not require a
    value. If a variable has a value there, it does require one.
 
-You may add configuration to the ``.custom_build_and_test`` job that will then
-be included to all you CI jobs. This can be used for example to `export jUnit
-test reports`_. Changes to that section are not mandatory.
+You may add configurations to the ``.custom_build_and_test`` job that will then
+be included in all you CI jobs. This can be used `export jUnit test reports`_,
```
Comment (adrienbernede) [inline docs/sphinx/user_guide/setup_ci.rst:197]:
```suggestion
be included in all you CI jobs. This can be used to `export jUnit test reports`_,
```

**Review by adrienbernede — COMMENTED** [review]
Thank you so much for doing this @rhornung67! Of course I accept your changes. I have only made minor changes and I will restore the Spack section and proofread it myself to fix the links.

**Pull request merged**